clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -9192,7 +9194,7 @@ ComputeDefaultedComparisonExceptionSpec(Sema &S, SourceLocation Loc,
     EnterExpressionEvaluationContext Context(
         S, Sema::ExpressionEvaluationContext::Unevaluated);
 
-    CXXRecordDecl *RD = cast<CXXRecordDecl>(FD->getLexicalParent());
+    auto RD = getRecordDeclFromFirstParameter(FD);
```

Comment:
Hmmm... Does `cast<CXXRecordDecl>(FD->getDeclContext())` work? That way, we can avoid the `getRecordDeclFromFirstParameter` whole thing.
I suspect the current approach isn't right here because 1. we have to ensure FD is a CXXMethodDecl; 2. the parameter list could be empty and thus we probably run into an out-of-bound error.

Reply:
@zyn0217 Thanks for the feedback and for pointing out that helper. I've updated the code to use it. Previously, I used a solution similar to `DefineDefaultedComparison` to maintain a consistent approach, which might make sense at the `DefineDefaultedComparison` stage but not in `ComputeDefaultedComparisonExceptionSpec`.